src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1094,6 +1335,14 @@ private void VisitPattern(BoundExpression expression, TypeWithState expressionRe
                             whenFalse = NullableFlowState.MaybeNull;
                             break;
                     }
+
+                    // Visit the pattern value so it has an inferred nullability
+                    // PROTOTYPE(nullable-api): should we have a way of skipping these "only for api" calls when there will be no rewriting of the nodes?
+                    var previousDisableDiagnostics = _disableDiagnostics;
+                    _disableDiagnostics = true;
```

Comment:
>_disableDiagnostics = true; [](start = 20, length = 27)

Just curious, what's a scenario where this is significant?

Reply:
If the "constant" pattern isn't a constant, and has some other nullability error in it.

---
In reply to: [265696774](https://github.com/dotnet/roslyn/pull/34017#discussion_r265696774) [](ancestors = 265696774)